src/main/java/org/opensearch/security/ssl/OpenSearchSecuritySSLPlugin.java

```diff
@@ -676,7 +678,7 @@ public List<String> getSettingsFilter() {
 
     @Override
     public Optional<SecureSettingsFactory> getSecureSettingFactory(Settings settings) {
-        return Optional.of(new OpenSearchSecureSettingsFactory(settings, sks, NOOP_SSL_EXCEPTION_HANDLER));
+        return Optional.of(new OpenSearchSecureSettingsFactory(threadPool, sks, NOOP_SSL_EXCEPTION_HANDLER, securityRestHandler));
```

Comment:
Can you quickly explain the movement of the threadpool to being passed here? I am not sure I follow the passing of it to the settings factory. I know you now create a Netty verifier for the HTTP handler but it seems like this was not done before so I am trying to follow how this has changed the factory class.

Reply:
Yes, sure, the Netty4 header verifier needs `threadpool` as an argument. Historically we have 2 plugins in security (I am not sure why but it is what it is), one inherent from other. So I moved `threadpool` to base plugin since it is needed by both of them now.